### PR TITLE
Build tooling QoL

### PR DESCRIPTION
**Pull request by sbchapin**
This PR brings in many changes that are stylistic and build-related, specifically...

- Direct support for **VSCode + Metals + BSP + Bloop** in the build project and the exterior project, without forgoing IntelliJ support
- **scalafmt support**, including usage of `scalafmtCheck` in the inner project, executed as a test on the outer g8 project.  There is now a sane scalafmt configuration as well, and I've applied it against all files in the project
- **CI via github actions!**  This exercises the full creation of a project via g8 on various versions of Mac OS, with various versions of the JDK, and with versions of scala 2.12.  It creates the full project, tests it (guaranteeing tests pass), and then applies scalafmt to ensure it all works.
- Note that several scala source & test files have changed, but they are **only formatting changes**.
- Lastly, I've removed the `maintainer` g8 parameter, as it was deprecated from Docker and removed in SBT.  Go with the flow.

### CODE REVIEW

File: src/main/g8/README.md [hunk 1]
```diff
@@ -152,6 +152,26 @@ Below are the libraries used to provide a broad starting base for this project.
...
+
+## [scalafmt](https://scalameta.org/scalafmt/) - *(automatic)* code formatting ##
+
+`scalafmt` is a scala code formatter. It integrates directly into the interpreter and can use the full power of scala's symbolic interpretation, enabling [all sorts of crazy configurations](https://scalameta.org/scalafmt/docs/configuration.html).  What's important here is consistency, so we enforce it with scalafmt.
+
+###### Why do we use it? ######
+
+- Automatic formatting means you just slap that save button and let the interpreter and scalafmt do the rest
+- Consistent formatting ensured, ruled by the iron fist of your CI system
+- Less arguing and nitpicking - your S.O. won't dump you because you use spaces rather than tabs
```
Comment (SwiftEngineer) [inline src/main/g8/README.md:165]: 😂

File: build.sbt [hunk 2]
```diff
@@ -1,14 +1,23 @@
...
+      "-Xms1024m",
+      "-Xmx1024m",
+      "-XX:ReservedCodeCacheSize=128m",
+      "-XX:MaxPermSize=256m",
+      "-Xss2m",
+      "-Dfile.encoding=UTF-8"
+    ),
+    resolvers += Resolver.url("typesafe", url("https://repo.typesafe.com/typesafe/ivy-releases/"))(
+      Resolver.ivyStylePatterns
+    ),
```
Comment (SwiftEngineer) [inline build.sbt]: I'm just now realizing that I'm not sure what we're fetching from this resolver 🤔 you think it's still needed?

File: README.md [hunk 3]
```diff
@@ -41,13 +41,13 @@ git clone git@github.com:{YOUR_NAME_HERE}/quickstart-scala-sbt-native-image.g8.g
...
-# Test it continuously: (will extract the project then run tests inside extracted project)
+# Test it continuously: (will extract the project then run tests inside extracted project as well as checking code formatting)
 sbt ~test
 
 # Change it:
 git checkout -b cool-feature-branch
-echo 'Sam sucks' >> README.md
-git commit -am 'Stating the evident'
+echo 'Cool feature' >> README.md
+git commit -am 'Added something!'
```
Comment (SwiftEngineer) [inline README.md:50]: I was thinking about updating these same lines, but didn't wanna let it go to your head 😂 glad you updated it

**Review by SwiftEngineer — APPROVED** [review]
this is pog 🎉

File: build.sbt (hunk 2 above)
Comment (sbchapin) [reply build.sbt]: Just removed it.  Gonna let CI tell me 😏

File: README.md (hunk 3 above)
Comment (sbchapin) [reply README.md:50]: Yeah, I figured there's no reason to state the obvious.  I was kind of flouting Grice's maxim of quantity here, and that's a bit of an antipattern.

File: build.sbt (hunk 2 above)
Comment (sbchapin) [reply build.sbt]: CI agrees with you.

**Pull request merged**